include/Gaffer/SplinePlug.h

```diff
@@ -72,10 +72,20 @@ class SplinePlug : public ValuePlug
 		IECORE_RUNTIMETYPED_DECLARETEMPLATE( SplinePlug<T>, ValuePlug );
 		IE_CORE_DECLARERUNTIMETYPEDDESCRIPTION( SplinePlug<T> );
 
+		// TODO - this is where John had suggested this go in his initial email, but does it
+		// really make for this enum to be part of each separate templated version of SplinePlug?
+		enum Interpolation
+		{
+			Linear,
+			CatmullRom,
+			BSpline,
+		};
+
 		SplinePlug(
 			const std::string &name = defaultName<SplinePlug>(),
 			Direction direction=In,
-			const T &defaultValue = T(),
+			const typename T::PointContainer &defaultPoints = T::PointContainer(),
```

Comment:
How are you feeling about using two separate variables for the default and value? I'm wondering if we shouldn't wrap them up in a simple little `SplinePlug::ValueType` struct. This would also fix the problem that the current `ValueType` typedef is inaccurate - advertising that `Spline` is the value type.